File: clinicaapp/src/main/java/com/clinicaapp/model/Producto.java
package com.clinicaapp.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "productos")
public class Producto {
    @Id
    private String id;
    private String nombre;
    private String descripcion;
    private Double precio;
    private Integer stock;
    private String imagenUrl;
    private String categoria;
    private String clinicaId;
    private String lote;
    private String fechaVencimiento;
    private Integer stockMinimo;

    public Producto() {}

    public Producto(String nombre, String descripcion, Double precio, Integer stock, String imagenUrl, String categoria, String clinicaId, String lote, String fechaVencimiento, Integer stockMinimo) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.stock = stock;
        this.imagenUrl = imagenUrl;
        this.categoria = categoria;
        this.clinicaId = clinicaId;
        this.lote = lote;
        this.fechaVencimiento = fechaVencimiento;
        this.stockMinimo = stockMinimo;
    }

    public Producto(String nombre, String descripcion, Double precio, Integer stock, String imagenUrl, String categoria) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.precio = precio;
        this.stock = stock;
        this.imagenUrl = imagenUrl;
        this.categoria = categoria;
    }
}

File: clinicaapp_mobile/lib/models/producto.dart
class Producto {
  final String id;
  final String nombre;
  final String descripcion;
  final double precio;
  final int stock;
  final String imagenUrl;
  final String categoria;

  Producto({
    required this.id,
    required this.nombre,
    required this.descripcion,
    required this.precio,
    required this.stock,
    required this.imagenUrl,
    required this.categoria,
  });

  factory Producto.fromJson(Map<String, dynamic> json) {
    return Producto(
      id: json['id'] ?? '',
      nombre: json['nombre'] ?? '',
      descripcion: json['descripcion'] ?? '',
      precio: (json['precio'] as num).toDouble(),
      stock: json['stock'] ?? 0,
      imagenUrl: json['imagenUrl'] ?? '',
      categoria: json['categoria'] ?? '',
    );
  }
}
